src/actors/workers/blocks.rs

```diff
@@ -85,27 +91,57 @@ where
         Ok(blocks)
     }
 
+    /// Collect blocks according to the predicate `fun` and send those blocks to
+    ///  the metadata actor.
+    async fn collect_and_send(&self, fun: impl Fn(u32) -> bool + Send + 'static) -> Result<()> {
+        self.meta
+            .send(BatchBlock::new(self.collect_blocks(fun).await?))
+            .await?;
+        Ok(())
+    }
+
     /// First run of indexing
     /// gets any blocks that are missing from database and indexes those.
     /// sets the `last_max` value.
-    async fn re_index(&mut self) -> Result<Option<Vec<Block<B>>>> {
+    async fn re_index(&mut self) -> Result<()> {
```

Comment:
Would it be useful to communicate back to the caller if we're done vs there are batches left to process?

Reply:
From my understanding, when `re_index` returns it should have completely caught up until `last_max` and thus there will not be any batches left to process.

Reply:
ofc, my bad. 👍